### ISSUE TITLE
squash and merge tags original branch not new target

### ISSUE THREAD

**Issue by nmichlo** [anothrNick/github-tag-action#215]
Not sure if I am missing an obvious config setting, but squashing and merging, instead of a plain merge tags the original commit on the old branch from a PR rather than the new squashed and merged commit on our main branch.

**Comment by sbe-arg** [comment]
I think you are right

**Comment by nmichlo** [comment]
Thank you for your response.

Figured out the solution in the meantime. It's to change the ref for the second config in the readme to ${{ github.sha }} instead of the reference to the PR's head sha

Maybe a note can be added there for future users? Seems to be working fine now.

**Comment by sbe-arg** [comment]
Updated example in this pr https://github.com/anothrNick/github-tag-action/pull/218

**Comment by nmichlo** [comment]
'''
...

name: version bump
on:
  pull_request:
    types:
      - closed
    branches:
      - main

jobs:
  build:
    if: github.event.pull_request.merged == true
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3
        with:
          ref: ${{ github.event.pull_request.merge_commit_sha }}
          fetch-depth: "0"

...
'''

After looking at the GitHub docs, this is probably the correct reference to use for PR. Not the original one I posted.

I also needed to add the of statement which makes sure that the PR is actually merged, and not only closed (cancelled). Previously it was tagging the wrong branch too when it was unnerved but just cancelled.
